Gems/Atom/Tools/ShaderManagementConsole/Code/Source/Window/ShaderManagementConsoleTableView.cpp

```diff
@@ -25,10 +51,42 @@ namespace ShaderManagementConsole
         setEditTriggers(QAbstractItemView::NoEditTriggers);
         setSelectionBehavior(QAbstractItemView::SelectItems);
         setSelectionMode(QAbstractItemView::SingleSelection);
-        horizontalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
+        verticalHeader()->setSectionResizeMode(QHeaderView::ResizeToContents);
+        horizontalHeader()->setSectionResizeMode(QHeaderView::Interactive);
+        setAlternatingRowColors(true);
 
         RebuildTable();
         AtomToolsFramework::AtomToolsDocumentNotificationBus::Handler::BusConnect(m_toolId);
+        this->setContextMenuPolicy(Qt::CustomContextMenu);
+        this->connect(this, &QTableWidget::customContextMenuRequested, this, &ShaderManagementConsoleTableView::ShowContextMenu);
+    }
+
+    void ShaderManagementConsoleTableView::mousePressEvent(QMouseEvent* e)
+    {
+        QTableWidget::mousePressEvent(e);
+        if (e->button() == Qt::RightButton)
+        {
+            ShowContextMenu(e->pos());
+        }
+    }
+
+    void ShaderManagementConsoleTableView::ShowContextMenu(const QPoint& pos)
+    {
+        QMenu contextMenu(tr("Context menu"), this);
+        QAction* action = new QAction(QString(tr("Add variant")), this);
+        connect(
+            action,
+            &QAction::triggered,
+            this,
+            [this]()
+            {
+                ShaderManagementConsoleDocumentRequestBus::Event(
+                    m_documentId,
+                    &ShaderManagementConsoleDocumentRequestBus::Events::AddOneVariantRow);
+            });
+        contextMenu.addAction(action);
+
+        contextMenu.exec(mapToGlobal(pos));
```

Comment:
If you connect to the bus and override the functions listed below you can add this action to the main menu edit menu instead of managing your own context menu. This will also give you the opportunity to add other actions with shortcut keys. Rather than having the extra row and offset logic for the delete button or the context menu for the add button, they could both be in the edit menu with standard shortcut keys.

        // AtomToolsMainMenuRequestBus::Handler overrides...
        void CreateMenus(QMenuBar* menuBar) override;
        void UpdateMenus(QMenuBar* menuBar) override;

There is a bit of trickery when you have multiple documents open. You just have to connect or disconnect to the bus when the selected document changes. Material canvas GraphDocumentView does this, listening to the document notification bus for document changes, disconnecting on most events and only reconnecting if the document you care about is being opened.

Reply:
I see. We can do something like that in a second PR